F/FMM3D/build_tarballs.jl

```diff
@@ -0,0 +1,113 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+
+# author: Travis Askham (askhamwhat@gmail.com)
+#
+#TODO: there are a number of stumbling points for the
+# FAST_KER=ON option when cross-compiling. it is completely
+# disabled for now. Some issues:
+# - on musl, darwin, and freebsd various standard libraries
+#   are not found
+# - on mingw there is an invalid regiser error on x86_64 arch
+# - generally, these seem to require a later version of gcc
+#   but we would like to compile for various libgfortran
+#   major versions
+
+using BinaryBuilder, Pkg
+
+name = "FMM3D"
+version = v"0.0.0"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/flatironinstitute/FMM3D.git", "2fd129984e48abe437fddd12013dc434563a773a")
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd FMM3D/
+
+touch make.inc
+
+echo "CC=${CC}" >> make.inc;
+echo "CXX=${CXX}" >> make.inc;
+echo "FC=${FC}" >> make.inc;
+
+echo "FFLAGS= -fPIC -O3 -march=native -funroll-loops" >> make.inc;
```

Comment:
You should not use -march=native; that will use CPU instructions that are specific to the machine the build is happening on, which will cause the binary to be non-portable. That's probably why this build gives off warnings like these: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=3871&view=logs&j=1bf8c0af-a6b0-52b0-ad08-e88c1bcbffd5&t=2991b1eb-2457-54ee-6715-263cf247349b&l=311

Allowing users to build multiple different tarballs split across microarchitectural boundaries (e.g. creating one that targets base `x86_64`, then one that targets `avx`, then one that targets `avx512`, etc...) is on our roadmap, but we aren't quite there yet.  For the time being, we only support two cases:

* Everything gets built for a single, conservative microarchitecture
* The library is smart enough, inside of itself, to do a `cpuid` and dispatch to the correct function.  Unfortunately, it appears that FMM3D does not have any `cpuid` instructions inside of it (if it did, BB would not give off the above warning) so you'll just have to live with the lower performance from not using AVX for now until we are able to push BB forward to the point that we are ready to compile even more configurations than we already are.